spec/commands/update_spec.rb

```diff
@@ -109,6 +129,96 @@
       expect(exitstatus).not_to eq(0) if exitstatus
     end
   end
+
+  describe "with --source option" do
+    it "should not update gems outwith the source that happen to have the same name" do
+      pending("Allowed to fail to preserve backwards-compatibility")
+
+      install_gemfile <<-G
+        source "file://#{gem_repo2}"
+        gem "activesupport"
+      G
+      update_repo2 { build_gem "activesupport", "3.0" }
+
+      bundle "update --source activesupport"
+      should_not_be_installed "activesupport 3.0"
+    end
+
+    it "should update gems outwith the source that happen to have the same name" do
```

Comment:
Ah, sorry — Scottish English. In this context it means 'gems not sourced from the provided source'. I'll amend for clarity (hopefully more clear than the example I gave).